### ISSUE TITLE
ROC-RK3588S-PC's fan alwas run at full speed

### ISSUE THREAD

**Issue by yangjianying** [Joshua-Riek/ubuntu-rockchip#642]
ROC-RK3588S-PC's fan alwas run at full speed, when boot up from tf card. if have time to handle this problem.

**Comment by mo123** [comment]
Hi

It was fixed in Armbian
https://github.com/armbian/linux-rockchip/commit/64df218de95ba41ad42c1fe7b310d874a1950eb7

@Joshua-Riek 
Can you check or change the dtb file for ROC-RK3588S-PC to the new one used by Armbian that supports fan speeds?

**Comment by stevepringle** [comment]
This is probably the same issue found on Ub untu with the Raspberry Pi 5 
https://bugs.launchpad.net/ubuntu/+source/linux-raspi/+bug/2041741

**Comment by Joshua-Riek** [comment]
> Hi
> 
> It was fixed in Armbian [armbian/linux-rockchip@64df218](https://github.com/armbian/linux-rockchip/commit/64df218de95ba41ad42c1fe7b310d874a1950eb7)
> 
> @Joshua-Riek Can you check or change the dtb file for ROC-RK3588S-PC to the new one used by Armbian that supports fan speeds?

Yes this commit is in the kernel, i do not have this board so I can not test.

**Comment by isolitude** [comment]
I checked and found it still use "rk3588s-roc-rk3588s-pc-v12.dtb" not "rk3588s-roc-pc.dtb" . how can I change it?

**Comment by mo123** [comment]
> I checked and found it still use "rk3588s-roc-rk3588s-pc-v12.dtb" not "rk3588s-roc-pc.dtb" . how can I change it?

Change it here
https://github.com/Joshua-Riek/ubuntu-rockchip/pull/493/files#diff-892f86dc6687587856c8702dcb9596393d622c1ff16db1ac22274f7c0aab7924R110

**Comment by isolitude** [comment]
Solved, I change "fdtfile=rk3588s-roc-rk3588s-pc-v12.dtb" to "fdtfile=rk3588s-roc-pc.dtb" in /boot/firmware/ubuntuEnv.txt
